src/main/java/org/xstefank/check/TemplateChecker.java

```diff
@@ -64,8 +62,14 @@ public void checkPR(JsonNode payload) {
 
         if (format.getCommit() != null) {
             checks.add(new LatestCommitCheck(format.getCommit()));
-        } else if (title != null)
-            checks.add(new LatestCommitCheck(title));
+        }
+
+        Pattern regex = format.getRegex();
```

Comment:
I am sorry, but this is by no means what I had in mind with regards to this issue. Your task is to allow having `title` defined (normal check for title is included) but it is not required that the commit message is the same (matches the same regex) as is defined for title (of course only if the format YAML doesn't contain `commit` tag with the same regex)